File: backend-csharp/Tests/MongoTestFixture.cs
using CollaborativeDashboard.Api.Models;
using CollaborativeDashboard.Api.Services;
using Microsoft.Extensions.Options;
using MongoDB.Driver;
using Testcontainers.MongoDb;

namespace CollaborativeDashboard.Api.Tests;

public sealed class MongoTestFixture : IAsyncLifetime
{
    private const string DatabaseName = "collaborative-dashboard-tests";
    private readonly MongoDbContainer container = new MongoDbBuilder()
        .WithImage("mongo:7.0")
        .Build();
    private IMongoDatabase database = null!;

    public MongoDbContext Context { get; private set; } = null!;

    public async Task InitializeAsync()
    {
        await container.StartAsync();
        var client = new MongoClient(container.GetConnectionString());
        database = client.GetDatabase(DatabaseName);
        Context = new MongoDbContext(
            client,
            Options.Create(new MongoDbSettings { DatabaseName = DatabaseName }));
    }

    public async Task ResetAsync()
    {
        await database.DropCollectionAsync("boards");
        await database.DropCollectionAsync("users");
        await database.DropCollectionAsync("tasks");
    }

    public Task DisposeAsync() => container.DisposeAsync().AsTask();
}

[CollectionDefinition("Mongo")]
public sealed class MongoCollection : ICollectionFixture<MongoTestFixture>;

File: backend-java/src/test/java/com/collaborative/dashboard/MongoTestFixture.java
package com.collaborative.dashboard;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
import org.junit.jupiter.api.extension.AfterEachCallback;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.ExtensionContext;
import org.testcontainers.containers.MongoDBContainer;

public final class MongoTestFixture implements BeforeAllCallback, AfterEachCallback {
    private static final MongoDBContainer CONTAINER = new MongoDBContainer("mongo:7.0");
    private static MongoClient client;
    private static MongoDatabase database;

    public static MongoDatabase database() {
        return database;
    }

    public static String connectionString() {
        return CONTAINER.getConnectionString();
    }

    @Override
    public void beforeAll(ExtensionContext context) {
        if (!CONTAINER.isRunning()) {
            CONTAINER.start();
            client = MongoClients.create(CONTAINER.getConnectionString());
            database = client.getDatabase("collaborative-dashboard-tests");
        }
    }

    @Override
    public void afterEach(ExtensionContext context) {
        database.getCollection("users").drop();
        database.getCollection("boards").drop();
        database.getCollection("tasks").drop();
    }
}